lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -41,14 +41,36 @@ DartType? getExpectedType(PostfixExpression node) {
       node.thisOrAncestorMatching((e) => e.parent is! ParenthesizedExpression);
   var parent = realNode?.parent;
 
+  DartType? Function(DartType? type) mayHandleAwait = (type) => type;
+  if (parent is AwaitExpression) {
+    mayHandleAwait =
+        (type) => (type as ParameterizedType?)?.typeArguments.first;
```

Comment:
humm, there was indeed issue with await handling. The `return await` case is the only one that need a special processing because the target type is really a Future (unlike other case where the unwrap type is provided)